File: human_class/Human.mjs
export default class Human {
	constructor(opt) {
		this.name = opt["name"];
        this.sex = opt["sex"];
    }

	sayName() {
		console.log("My name is " + this.name);
    }

	saySex() {
		console.log("My sex is " + this.sex);
    }
};

if(!module.parent) {
	let human1 = new Human({"name":"FirstName LastName", "sex":"male"});
	human1.sayName();
	human1.saySex();
}

File: examples/human_class/Human.rb
class Human
    def initialize(opt)
        @name = opt["name"]
        @sex = opt["sex"]
    end

    def sayName
        puts "My name is " + @name.to_s
    end

    def saySex
        puts "My sex is " + @sex.to_s
    end
end

if $0 == __FILE__
    person1 = Human.new({"name" => "FirstName LastName", "sex" => "male"})
    person1.sayName
    person1.saySex
end
